test/jdk/java/awt/print/PrinterJob/PrintLatinCJKTest.java

```diff
@@ -23,55 +23,60 @@
 
 /**
  * @test
- * @bug 800535 8022536
+ * @bug 8022536
+ * @library ../../regtesthelpers
+ * @build Util
  * @summary JDK7 Printing: CJK and Latin Text in string overlap
- * @run main/manual=yesno PrintLatinCJKTest
+ * @run main/manual PrintLatinCJKTest
  */
 
 import java.awt.Font;
 import java.awt.Graphics;
-import java.awt.event.ActionEvent;
-import java.awt.event.ActionListener;
 import java.awt.print.PageFormat;
-import java.awt.print.Pageable;
 import java.awt.print.Printable;
 import java.awt.print.PrinterException;
 import java.awt.print.PrinterJob;
+import java.lang.reflect.InvocationTargetException;
 import javax.swing.JButton;
 import javax.swing.JFrame;
-import javax.swing.JTextArea;
-
 import javax.swing.SwingUtilities;
 
-public class PrintLatinCJKTest implements Printable, ActionListener {
+public class PrintLatinCJKTest implements Printable {
 
     static PrintLatinCJKTest testInstance = new PrintLatinCJKTest();
-    private PageFormat pf;
-
     static String info =
-       "To test 8022536, if a remote printer is the system default,"+
+       "To test 8022536, if a remote printer is the system default, \n"+
        "it should show in the dialog as the selected printer.\n"+
-       "You need a printer for this test. If you have none, let "+
-       "the test pass. If there is a printer, press Print, send "+
-       "the output to the printer, and examine it. It should have "+
+       "You need a printer for this test. If you have none, let \n"+
+       "the test pass. If there is a printer, press Print, send \n"+
+       "the output to the printer, and examine it. It should have \n"+
        "text looking like this : \u4e00\u4e01\u4e02\u4e03\u4e04English.";
 
-    public static void showFrame() {
-         JFrame f = new JFrame();
-         JTextArea jta = new JTextArea(info, 4, 30);
-         jta.setLineWrap(true);
-         jta.setWrapStyleWord(true);
-         f.add("Center", jta);
-         JButton b = new JButton("Print");
-         b.addActionListener(testInstance);
-         f.add("South", b);
-         f.pack();
-         f.setVisible(true);
+    public static void showFrame() throws InterruptedException, InvocationTargetException {
+        SwingUtilities.invokeAndWait( () -> {
+            JFrame f = new JFrame("Test Frame");
+            JButton b = new JButton("Print");
+            b.addActionListener((ae) -> {
+                try {
+                    PrinterJob job = PrinterJob.getPrinterJob();
+                    job.setPrintable(testInstance);
+                    if (job.printDialog()) {
+                        job.print();
+                    }
+                } catch (PrinterException ex) {
+                    ex.printStackTrace();
+                }
+            });
+            f.add("South", b);
+            f.pack();
+            f.setLocationRelativeTo(null);
+            f.setVisible(true);
+        });
```

Comment:
I don't see where/how this frame is disposed.

Reply:
Added code to dispose the test UI.